.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,29 @@
+// For format details, see https://aka.ms/vscode-remote/devcontainer.json or this file's README at:
+// https://github.com/microsoft/vscode-dev-containers/tree/v0.112.0/containers/cpp
+{
+	"name": "C++",
+	"dockerFile": "Dockerfile",
+	// Due to a limitation in podman, the development container should be executed
+	// in "privileged" mode: https://github.com/containers/libpod/issues/4131
+	"runArgs": [
+		"--privileged"
+	],
+	// "runArgs": [ "--cap-add=SYS_PTRACE", "--security-opt", "seccomp=unconfined" ],
+	// Set *default* container specific settings.json values on container create.
+	"settings": {
+		"terminal.integrated.shell.linux": "/bin/bash"
+	},
+	// Expose container ports
+	"forwardPorts": [
+		2222
+	],
+	// IDs of extensions to install inside the container
+	"extensions": [
+		"ms-vscode.cpptools",
+		"twxs.cmake",
+		"eamodio.gitlens",
+		"github.vscode-pull-request-github",
+		"ms-python.python",
+		"redhat.vscode-yaml"
```

Comment:
This is great! I would also recommend adding:
spellcheck (timonwong.shellcheck)
Markdown Preview Enhanced (shd101wyy.markdown-preview-enhanced)
Doxygen Documentation Generator (cschlosser.doxdocgen)
Code Spell Checker (streetsidesoftware.code-spell-checker) - Hopefully we will have less typo error with this plugin

Reply:
Added the Code Spell Checker and a Markdown All in One extensions